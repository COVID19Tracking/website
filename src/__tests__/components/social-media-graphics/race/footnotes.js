import React from 'react'
import renderer from 'react-test-renderer'
import Footnotes from '~components/social-media-graphics/race/footnotes'

const combined = {
  knownRaceEthPos: 0.98,
  knownRaceEthDeath: 0.43,
}

const separate = {
  knownRacePos: 0.32,
  knownRaceDeath: 0.76,
  knownEthPos: 0.99,
  knownEthDeath: 1,
}

describe('Components : Social Media Graphics : Race : Footnotes : Utah', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Footnotes
          state={combined}
          stateName="Utah"
          asteriskFootnote="Here is an additional footnote"
          showSmallNFootnote
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('Components : Social Media Graphics : Race : Footnotes : Wyoming', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Footnotes state={separate} stateName="Wyoming" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('Components : Social Media Graphics : Race : Footnotes : National', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Footnotes
          state={separate}
          stateName="United States"
          statesReportingDeaths={34}
          statesReportingCases={49}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('Components : Social Media Graphics : Race : Footnotes : Generic separate', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Footnotes state={separate} stateName="Alabama" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('Components : Social Media Graphics : Race : Footnotes : Generic combined', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Footnotes state={combined} stateName="Alabama" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
