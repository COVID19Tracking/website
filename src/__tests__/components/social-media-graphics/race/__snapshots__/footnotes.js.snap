// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components : Social Media Graphics : Race : Footnotes : Generic combined renders correctly 1`] = `
<div
  className="notes"
>
  <div>
    <strong>
      Notes:
    </strong>
     
    Alabama
     has reported race and ethnicity data for 
    98
    %
     of cases and
     
    43
    %
     of deaths. Graphic only includes demographic groups reported by the state. Race categories are mutually exclusive and defined as not Hispanic or Latino.
  </div>
</div>
`;

exports[`Components : Social Media Graphics : Race : Footnotes : Generic separate renders correctly 1`] = `
<div
  className="notes"
>
  <div>
    <strong>
      Notes:
    </strong>
     
    Alabama
     has reported race data for
     
    32
    %
     of cases and
     
    76
    %
     of deaths, and ethnicity data for 
    99
    %
     of cases and
     
    100
    %
     of deaths. Graphic only includes demographic groups reported by the state. Race categories are mutually exclusive and include both Hispanic/Latino and non-Hispanic/Latino ethnicity.
  </div>
</div>
`;

exports[`Components : Social Media Graphics : Race : Footnotes : National renders correctly 1`] = `
<div
  className="notes national"
>
  <div>
    <strong>
      Notes:
    </strong>
     Nationwide,
     
    49
     of 56 states and territories report race/ethnicity information for cases and
     
    34
     of 56 report race/ethnicity for deaths. Graphic includes demographic data from all states and territories that report, using standard Census categories where possible, and scaled to the total US population for each Census category. Race categories may overlap with Hispanic/Latino ethnicity. Some rates are underestimated due to lack of reporting of race and ethnicity categories for COVID-19 cases and deaths.
  </div>
</div>
`;

exports[`Components : Social Media Graphics : Race : Footnotes : Utah renders correctly 1`] = `
<div
  className="notes"
>
  <div>
    <strong>
      Notes:
    </strong>
     Utah has reported race and ethnicity data for
     
    98
    %
     of cases and
     
    43
    %
     of deaths. Graphic only includes demographic groups reported by the state. Race categories are non-mutually-exclusive and are defined as not Hispanic or Latino.
  </div>
  <div
    className="asteriskFootnote"
  >
    * Based on fewer than 10 deaths among members of this race/ethnicity. Interpret with caution.
    <br />
    â€  
    Here is an additional footnote
  </div>
</div>
`;

exports[`Components : Social Media Graphics : Race : Footnotes : Wyoming renders correctly 1`] = `
<div
  className="notes"
>
  <div>
    <strong>
      Notes:
    </strong>
     Wyoming has reported race data for
     
    32
    %
     of cases and
     
    76
    %
     of deaths, and ethnicity data for 
    99
    %
     of cases and
     
    100
    %
     of deaths. Graphic only includes demographic groups reported by the state. Race categories are non-mutually-exclusive and include both Hispanic/Latino and non-Hispanic/Latino ethnicity.
  </div>
</div>
`;
